Basic-Car-Maintenance/Shared/Odometer/ViewModels/OdometerViewModel.swift

```diff
@@ -36,6 +36,28 @@ class OdometerViewModel {
         }
     }
     
+    func deleteReading(_ reading: OdometerReading) async {
+        guard let documentId = reading.id else {
+            fatalError("Reading Entry has no document ID.")
+        }
+        
+        do {
+            try await Firestore
+                .firestore()
+                .collection(FirestoreCollection.odometerReadings)
+                .document(documentId)
+                .delete()
+            errorMessage = ""
+            
+            if let eventIndex = readings.firstIndex(of: reading) {
+                readings.remove(at: eventIndex)
+            }
+        } catch {
+            errorMessage = error.localizedDescription
+            showAddErrorAlert = true
```

Comment:
The `showAddErrorAlert` doesn't make sense to use in this case, because it's only presented on the `AddOdometerReadingView`

Otherwise, everything else works

Reply:
Thanks for clarifying!